### ISSUE TITLE
Top bar icon to toggle the menu? To avoid having to context switch to the keyboard only to press Alt

### ISSUE THREAD

**Issue by mxmilkiib** [mixxxdj/mixxx#13266]
### Feature Description

I like the fact that the menu can be hidden now to save UI space.

It can then be shown by pressing Alt, but, if primaraly navigating the UI with the mouse, it is somewhat frustrating to have to reach for the keyboard just to press Alt.

How about a top bar icon that can also toggle the visibility of the menu?

**Comment by ronso0** [comment]
So you need the menu bar but also like to hide it : )

There was #3184, closed in favor of #3189, we might revive that.

For the fullscreen case we could configure a 'hot zone' at the top edge i.e. reveal the menu if the mouse remains at the 0-2 top pixels for a second or something. (not sure though, might conflict with other system bars / widgets)

**Comment by mxmilkiib** [comment]
I wish to get into the Preferences a lot, but hitting ctrl-p also requires the same slowing context switch!

**Comment by ronso0** [comment]
Yeah, a button for the preferences would be nice (like it exists? in QML)

For skins with a skin menu, we may add two toggles:
* show preferences
* show menu bar

**Comment by mxmilkiib** [comment]
A skin menu solution would take too many steps imho.

Click skin menu, click enable bar, click options, click preferences, *do stuff, close prefs*, click disable bar, click to close the skin menu: 6 clicks total.

Skin button would act more like pressing Alt; click button, click options, click prefs, *ting, close*: 3 clicks total.

Skin menu "show preferences"? How would that work? A toggle that brings up the prefs and autotoggles back on closing prefs? No.. A "button" like "Load Bank"? That's one extra click.

A skin button for opening the preferences would be quickest. One [also] to show the bar would provide [access to] utility. I guess I'm more interested in the former.

But look at all this whitespace;

![image](https://github.com/mixxxdj/mixxx/assets/108225/1ddad2b1-78bc-4271-847f-a08543dcdd0b)

I ultimately view that as wasted space. Prefs access is a major action, done much more for the majority of users compared to connecting to a remote streaming server, and that has an icon. Viewed in that "disproportionate" light, it's a no-brainer, if I might be so rhetorically bold.

**Comment by ronso0** [comment]
I totally agree one-click solutions would are preferred.

Thing is
> I ultimately view that as wasted space

This is only true for wide screens, though toolbars are already pretty crammed at min. skin size IMO (see #13244), especially in Tango.

**Comment by mxmilkiib** [comment]
bump, as #13301 was merged

if 'how to gracefully degrade when the full set of elements overlaps the allowed width' is the blocker;

simplest;

just add it, if someone complains then make the new button auto hide? better than nothing

or

partial label reduction;

there's room for the label text to truncate; "Spl.", "Mix", "FX", "Sam.", "Ste." "M & A"?

and / or

skin settings to make all bar elements just be icons?

and / or

an automatic/responsive approach would allow media query style techniques

.

does the app UI approach have a philosophy/view yet? i.e. would it be all/often buttons or responsive themes?

(like opening/hiding the beatgrid/bpm set of buttons, you could make a [dock](https://en.wikipedia.org/wiki/Dock_(macOS))/wharf with [drawers](https://www.dockapps.net/wmdrawer) (nested bars))

**Comment by ronso0** [comment]
> does the app UI approach have a philosophy/view yet?

No, unfortunately we have no explicit guideline, the skins have grown independently.
We don't have translatable buttons (that would justify making buttons responsive, like set a max width and dynamically elide labels) and a docks or drawers are out of scope for the legacy UI.

To move on here I suggest to add a Pref button.
I already have it in my custom build (e1cfea4b987f7895af5973bc8d033d329a39d281). I made the MIXXX logo toggle the skin settings and the ⚙️ toggle now opens the preferences. This has the downside that the former is not discoverable.
